microsoft/electionguard-ui
# 🔐 Authentication and Login Page

[//]: # (🚨 Please review the CONTRIBUTING.md in this repository. 💔Thank you!)

### Issue

Fixes #108 

### Description
Primarily this adds a login screen and meets the acceptance criteria in the ticket e.g.:

- If I put in invalid credentials, I should see a message that says they're invalid
- If I originally navigated via link to a more specific page, after login I want to see that page rather than the home screen
- If I’ve logged in previously, I still have to log in again if I close the tab
- I can’t view any page in the site without logging in
- I want the ability to log out

As a bonus this also implements the ElectionGuard teal theme in the UI and adds a logo to the title bar.  

Implementaiton-wise this adds nswag, which generates TypeScript API proxy classes for hitting the back-end.  We should plan to phase out the old manually written code as we phase in code that's generated from the swagger file.

### Testing

1. Navigate to any page in the site.  Expected: it should show a login screen.
2. Enter invalid credentials and click the button.  Expected: it should show an error about an invalid login attempt.
3. Enter valid credentials and click the button.  Expected: it should navigate into the site.
4. Click log out in the top right.  Expected: it should navigate back to the login screen.
